src/ZendCodingStandard/ruleset.xml

```diff
@@ -201,232 +70,823 @@
-    <!-- Require use of short versions of scalar types (i.e. int instead of integer) -->
-    <rule ref="SlevomatCodingStandard.TypeHints.LongTypeHints"/>
-    <!-- Require ? when default value is null -->
-    <rule ref="SlevomatCodingStandard.TypeHints.NullableTypeForNullDefaultValue"/>
-    <!-- Disallow space after nullable operator -->
-    <rule ref="PSR12.Functions.NullableTypeDeclaration"/>
-    <!-- Require one space between typehint and variable, require no space between nullability sign and typehint -->
-    <rule ref="SlevomatCodingStandard.TypeHints.ParameterTypeHintSpacing"/>
-    <!-- Require space around colon in return types -->
-    <rule ref="WebimpressCodingStandard.Formatting.ReturnType">
-        <!-- Covered by Generic.PHP.LowerCaseType -->
-        <exclude name="WebimpressCodingStandard.Formatting.ReturnType.LowerCaseSimpleType"/>
-    </rule>
-    <!-- Forbid useless variables -->
-    <rule ref="SlevomatCodingStandard.Variables.UselessVariable"/>
-    <!-- Forbid class being in a file with different name -->
+    <!-- Classes, traits, interfaces, constants and functions MUST be imported. -->
+    <rule ref="WebimpressCodingStandard.PHP.DisallowFqn"/>
+    <!-- Internal functions MUST be imported. -->
+    <rule ref="WebimpressCodingStandard.PHP.ImportInternalFunction"/>
+    <!-- Internal constants MUST be imported. -->
+    <rule ref="WebimpressCodingStandard.PHP.ImportInternalConstant"/>
+
+    <!-- 4. Classes, Properties, and Methods -->
+
+    <!-- Any closing brace MUST NOT be followed by any comment or statement on
+         the same line. -->
+    <!-- checked by PSR12.Classes.ClosingBrace -->
+
+    <!-- When instantiating a new class, parentheses MUST always be present
+         even when there are no arguments passed to the constructor. -->
+    <!-- checked by PSR12.Classes.ClassInstantiation -->
+
+    <!-- Additional Zend Framework rules -->
+
+    <!-- There MAY NOT be duplicate class names. -->
+    <rule ref="Generic.Classes.DuplicateClassName"/>
+    <!-- The file name MUST match the case of the terminating class name. -->
     <rule ref="Squiz.Classes.ClassFileName"/>
-    <!-- Force `self::` for self-reference, force lower-case self, forbid spaces around `::` -->
+    <!-- PHP 4 style constructors SHOULD NOT be used. -->
+    <rule ref="Generic.NamingConventions.ConstructorName"/>
+    <!-- Abstract classes MUST have a `Abstract` prefix. -->
+    <rule ref="WebimpressCodingStandard.NamingConventions.AbstractClass"/>
+    <!-- Exception classes MUST have a `Exception` suffix. -->
+    <rule ref="WebimpressCodingStandard.NamingConventions.Exception"/>
+    <!-- Interface classes MUST have a `Interface` suffix. -->
+    <rule ref="WebimpressCodingStandard.NamingConventions.Interface"/>
+    <!-- Trait classes MUST have a `Trait` suffix. -->
+    <rule ref="WebimpressCodingStandard.NamingConventions.Trait"/>
+    <!-- The correct class names MUST be used. -->
```

Comment:
What does this mean, exactly?

Reply:
Looking at the related sniff I think it means if a class is called `ZendFrameworkExpressiveException` that exact name must be used and not `ZendframeworkExpressiveexception`.

/cc @michalbundyra please confirm

Reply:
@geerteltink correct

Reply:
Confirmed that it does exactly that: e.g. `Datetime` -> `DateTime`.